### PR TITLE
CHERRY_PICK: [TCBZ3541] MinPlatformPkg/TestPointCheckLib: Add support for BME device exemption

### PR DESCRIPTION
**Pull request by makubacki**
REF:https://bugzilla.tianocore.org/show_bug.cgi?id=3541

Some platforms have devices which do not expose any additional
risk of DMA attacks but the BME bit cannot be disabled.

To allow MinPlatformPkg consumers to selectively exempt certain
devices from the PCI bus master test point, this change adds a
PCD to MinPlatformPkg.dec that allows those packages to specify
a list of PCI devices by S/B/D/F that should be excluded from
testing.

Cc: Chasel Chiu <chasel.chiu@intel.com>
Cc: Nate DeSimone <nathaniel.l.desimone@intel.com>
Cc: Liming Gao <gaoliming@byosoft.com.cn>
Cc: Eric Dong <eric.dong@intel.com>
Cc: Chris Ruffin <v-chruff@microsoft.com>
Co-authored-by: Michael Kubacki <michael.kubacki@microsoft.com>
Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>
(cherry picked from commit a9594e6)

### CODE REVIEW

File: MinPlatformPkg/Test/Library/TestPointCheckLib/DxeCheckPci.c [hunk 1]
```diff
@@ -256,7 +265,7 @@ TestPointCheckPciResource (
   UINT16                            MinBus;
   UINT16                            MaxBus;
   BOOLEAN                           IsEnd;
-  
+
```
Comment (corthon) [inline MinPlatformPkg/Test/Library/TestPointCheckLib/DxeCheckPci.c:268]: Unnecessary changes?

**Review by corthon — CHANGES_REQUESTED** [review]
If this is truly a TCBZ commit (as the comments suggest) we need to make sure that TCBZ#### is in the commit title so that the scripts pick it up.

File: MinPlatformPkg/Test/Library/TestPointCheckLib/DxeCheckPci.c (hunk 1 above)
Comment (makubacki) [reply MinPlatformPkg/Test/Library/TestPointCheckLib/DxeCheckPci.c:268]: It is just whitespace removal which should not have been allowed in the file in the first place. My editor is configured to automatically remove extra whitespace. I can undo it but it is also removed in the upstream patch.

Comment (makubacki) [reply MinPlatformPkg/Test/Library/TestPointCheckLib/DxeCheckPci.c:268]: Okay removed? Or add it back?
